### ISSUE TITLE
unrecognised option -t/--targets

### ISSUE THREAD

**Issue by molwizard** [COMBINE-lab/salmon#343]
**Is the bug primarily related to salmon (bulk mode) or alevin (single-cell mode)?**
salmon

**Describe the bug**
The online documentation at "https://salmon.readthedocs.io/en/latest/salmon.html#quantifying-in-alignment-based-mode" for using salmon to quantify data from another alignment program by using a BAM format file describes the option '-t transcripts.fa' for this purpose:

> ./bin/salmon quant -t transcripts.fa -l <LIBTYPE> -a aln.bam -o salmon_quant

Also the command-line help 'salmon quant --help-alignment' names the same options:

alignment input options:

  -l [ --libType ] arg                  Format string describing the library 
                                        type
  -a [ --alignments ] arg               input alignment (BAM) file(s).
  -t [ --targets ] arg                  FASTA format file containing target 
                                        transcripts.

The 'salmon quant' command doesn't recognize either the short or long form of this option and stops execution immediately. There is no difference if the full set of options for a quantification run are provided or only this option. 

**To Reproduce**
salmon quant -t Homo_sapiens.GRCh37.87.cdna.all.fa
salmon quant --targets Homo_sapiens.GRCh37.87.cdna.all.fa

Specifically, please provide at least the following information:

  * Which version of salmon was used?
0.12.0 and 0.11.3
  * How was salmon installed (compiled, downloaded executable, through bioconda)?
downloaded executable (salmon-0.12.0_linux_x86_64.tar.gz, salmon-0.11.3-linux_x86_64.tar.gz)
  * Which reference (e.g. transcriptome) was used?
any, independant of transcriptome
  * Which read files were used?
not applicable
  * Which which program options were used?
other options than '-t/--targets' were irrelevant for provoking the error 

**Expected behavior**
Salmon should run and process the alignment file in BAM format.

**Screenshots**
--- Error messages ------------------
Exception : [unrecognised option '-t']. Exiting.
Exception : [unrecognised option '--targets']. Exiting.
--------------------------------------

**Desktop (please complete the following information):**
 - OS: Ubuntu Linux
 - Version:
Linux nnn 4.15.0-36-generic #39~16.04.1-Ubuntu SMP Tue Sep 25 08:59:23 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.4 LTS
Release:	16.04
Codename:	xenial

**Additional context**
none

**Comment by rob-p** [comment]
Hi @molwizard,

  The problem seems to be that you are not providing the full command line.  If you are using alignment-based mode, you must provide the BAM file via the `-a` option. The library type via `-l` and the output folder via `-o` in addition to the transcriptome at the least.  In fact, salmon parses the command line after quant looking for `-a` to determine which mode to run in (alignment vs. mapping based).

**Comment by molwizard** [comment]
Hi @rob-p,

thank you for your quick answer. As mentioned already, I started with the full set of options (at least I thought so) and then reduced them to the minimal case to reproduce the error for reporting the problem.
I was mislead by the term 'unrecognized option' and didn't expect the program to "forget" options from other modes. But now that you stated that this is the case, I realized that the '-a' in front of the BAM file name was missing, which I overlooked before. After adding it the program at least started to run. (Although it ran into another crirical error, appearantly misinterpreting chromosome and supercontig names from the BAM file header generated by STAR as transcript names.)

Maybe it would be a good idea to distinguish in the error messages between 'unrecognised' and 'inappropriate' options to provide a better clue to the user what went wrong.
